tests/store/model_registry/test_rest_store.py

```diff
@@ -33,327 +32,353 @@
-    def test_get_latest_versions_with_stages(self, mock_multiple_http_requests):
-        name = "model_1"
-        self.store.get_latest_versions(name=name, stages=["blaah"])
-        endpoint = "registered-models/get-latest-versions"
-        endpoints = [(endpoint, "POST"), (endpoint, "GET")]
-        self._verify_all_requests(
-            mock_multiple_http_requests, endpoints, GetLatestVersions(name=name, stages=["blaah"])
-        )
 
-    @mock_http_request
-    def test_set_registered_model_tag(self, mock_http):
-        name = "model_1"
-        tag = RegisteredModelTag(key="key", value="value")
-        self.store.set_registered_model_tag(name=name, tag=tag)
-        self._verify_requests(
-            mock_http,
-            "registered-models/set-tag",
-            "POST",
-            SetRegisteredModelTag(name=name, key=tag.key, value=tag.value),
-        )
+def test_delete_registered_model(store, creds):
+    name = "model_1"
+    with mock_http_request_200() as mock_http:
+        store.delete_registered_model(name=name)
+    _verify_requests(
+        mock_http, creds, "registered-models/delete", "DELETE", DeleteRegisteredModel(name=name)
+    )
 
-    @mock_http_request
-    def test_delete_registered_model_tag(self, mock_http):
-        name = "model_1"
-        self.store.delete_registered_model_tag(name=name, key="key")
-        self._verify_requests(
-            mock_http,
-            "registered-models/delete-tag",
-            "DELETE",
-            DeleteRegisteredModelTag(name=name, key="key"),
-        )
 
-    @mock_http_request
-    def test_create_model_version(self, mock_http):
-        self.store.create_model_version("model_1", "path/to/source")
-        self._verify_requests(
-            mock_http,
-            "model-versions/create",
-            "POST",
-            CreateModelVersion(name="model_1", source="path/to/source"),
-        )
-        # test optional fields
-        run_id = uuid.uuid4().hex
-        tags = [
-            ModelVersionTag(key="key", value="value"),
-            ModelVersionTag(key="anotherKey", value="some other value"),
-        ]
-        run_link = "localhost:5000/path/to/run"
-        description = "version description"
-        self.store.create_model_version(
+def test_search_registered_models(store, creds):
+    with mock_http_request_200() as mock_http:
+        store.search_registered_models()
+    _verify_requests(mock_http, creds, "registered-models/search", "GET", SearchRegisteredModels())
+
+
+def _search_registered_models_params():
+    params_list = [
+        {"filter_string": "model = 'yo'"},
+        {"max_results": 400},
+        {"page_token": "blah"},
+        {"order_by": ["x", "Y"]},
+    ]
+    # Generate all combinations of the above params
+    for sz in [0, 1, 2, 3, 4]:
+        for combination in combinations(params_list, sz):
+            params = {k: v for d in combination for k, v in d.items()}
+            yield params
```

Comment:
Can we use `@pytest.mark.parametrize` instead to generate the combinations ?

Reply:
This function is not a test. See line 163.

Reply:
I think we can.

Reply:
I think we can, each param we can set 2 options [some_value, None], None means we don't set the param when calling related api

Reply:
See https://github.com/mlflow/mlflow/pull/7780/commits/a7f8a024a1cf931c36de4a0bf363cba6dc7158f0